File: twilight-api/src/main/java/dev/bolko/twilightapi/model/Community.java
package dev.bolko.twilightapi.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
public final class Community {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 60)
    private String name;

    @Column(length = 500)
    private String description;

    private String image;

    @OneToMany(mappedBy = "community", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnore
    @OrderBy("createdAt DESC")
    private List<Post> posts = new ArrayList<>();

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "creator_id", nullable = false)
    @JsonBackReference
    private User creator;

    @ManyToMany(mappedBy = "communities")
    @JsonIgnore
    private Set<User> members = new HashSet<>();

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    public Community(String name, String description, String image, User creator) {
        this.name = name;
        this.description = description;
        this.image = image;
        this.creator = creator;
    }
    // --- helpers (používaj v service) ---
    public void addMember(User user) {
        members.add(user);
        user.getCommunities().add(this);
    }

    public void removeMember(User user) {
        members.remove(user);
        user.getCommunities().remove(this);
    }
}

File: server/Models/Community.cs
using System.ComponentModel.DataAnnotations;

namespace server.Models;

public class Community
{
    public Guid Id { get; set; }

    [Required]
    [MaxLength(60)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(500)]
    public string? Description { get; set; }

    public string? Image { get; set; }

    public string CreatorId { get; set; } = string.Empty;
    public ApplicationUser Creator { get; set; } = null!;

    public int? CategoryId { get; set; }
    public Category? Category { get; set; }

    public ICollection<Post> Posts { get; set; } = new List<Post>();

    public ICollection<ApplicationUser> Members { get; set; } = new List<ApplicationUser>();
    public ICollection<ApplicationUser> NightOwls { get; set; } = new List<ApplicationUser>();
}

